answers/ch04.ts

```diff
@@ -75,8 +75,12 @@ is(10, 'foo') // エラー TS2345: 型 '"foo"' の引数を型 'number' の
               // パラメーターに割り当てることはできません。
 
 // ［難問］任意の数の引数を渡せるようにします
+is([1], [1], [1]) // true
 is([1], [1, 2], [1, 2, 3]) // false
 
 function is<T>(a: T, ...b: [T, ...T[]]): boolean {
+  if (typeof a === 'object') {
+    return b.every(_ => JSON.stringify(_) === JSON.stringify(a))
```

Comment:
Thanks for your proposal!
However, using `JSON.stringify()` is problematic when the argument can be any JavaScript Object.

```ts
// ex. 1: JSON.stringify(regExp) always returns '{}'
JSON.stringify(/abc/) === JSON.stringify(/edf/) // true

// ex. 2: JSON.stringify() does not support circular structure
const a = {}
a.a = a
JSON.stringify(a) // Uncaught TypeError: Converting circular structure to JSON
```

Implementing a strict object comparison algorithm is apparently off-topic here, so I would suggest simply modifying the misleading example usage of `is([1], [1, 2], [1, 2, 3])` to something like `is(1, 2, 3)`.
What do you think?

Reply:
That is reasonable. I changed examples.